File: web/lib/schemas/notification.ts
import { z } from 'zod'

export const createNotificationSchema = z.object({
  user_id: z.string().uuid(),
  title: z.string().min(1).max(255),
  message: z.string().min(1),
  type: z.enum(['info', 'warning', 'success', 'error']),
  channel: z.enum(['in_app', 'line', 'email']),
  action_url: z.string().url().optional(),
})

export type CreateNotificationInput = z.infer<typeof createNotificationSchema>

File: common-api/internal/model/notification.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type Notification struct {
	ID        uuid.UUID  `json:"id"`
	UserID    uuid.UUID  `json:"user_id"`
	Title     string     `json:"title"`
	Message   string     `json:"message"`
	Type      string     `json:"type"`
	Channel   string     `json:"channel"`
	ReadAt    *time.Time `json:"read_at,omitempty"`
	ActionURL *string    `json:"action_url,omitempty"`
	CreatedAt time.Time  `json:"created_at"`
}

type CreateNotificationRequest struct {
	UserID    string `json:"user_id" binding:"required,uuid"`
	Title     string `json:"title" binding:"required"`
	Message   string `json:"message" binding:"required"`
	Type      string `json:"type" binding:"required,oneof=info warning success error"`
	Channel   string `json:"channel" binding:"required,oneof=in_app line email"`
	ActionURL string `json:"action_url,omitempty"`
}
